src/lib.rs

```diff
@@ -449,13 +479,21 @@ where
         }
     }
 
-    fn put_back(&self, mut conn: Conn<M::Connection>) {
+    fn put_back(&self, conn: &mut PooledConnection<M>) {
+        self.0.manager.before_return(conn);
```

Comment:
Maybe this should only be called if the connection is not broken (e.g. before line 498) so that the caller does not need to check for a broken connection as well